staking-pool/src/lib.rs

```diff
@@ -90,358 +111,514 @@ impl Default for StakingContract {
-            account
-        } else {
-            Account::new()
-        }
+    /// Returns the total staking balance.
+    pub fn get_total_staked_balance(&self) -> U128 {
+        self.total_staked_balance.into()
     }
 
-    fn pull_account(&mut self, account_id: AccountId) -> WrappedAccount {
-        if let Some(account) = self.active_accounts.remove(&account_id) {
-            WrappedAccount {
-                account_id,
-                account,
-                status: AccountStatus::Active,
-            }
-        } else if let Some(account) = self.inactive_accounts.remove(&account_id) {
-            WrappedAccount {
-                account_id,
-                account,
-                status: AccountStatus::Inactive,
-            }
-        } else if let Some(account) = self.archived_accounts.remove(&account_id) {
-            WrappedAccount {
-                account_id,
-                account,
-                status: AccountStatus::Inactive,
-            }
-        } else {
-            WrappedAccount {
-                account_id,
-                account: Account::new(),
-                status: AccountStatus::Inactive,
-            }
+    /// Returns account ID of the staking pool owner.
+    pub fn get_owner_id(&self) -> AccountId {
+        self.owner_id.clone()
+    }
+
+    /// Returns the current reward fee as a fraction.
+    pub fn get_reward_fee_fraction(&self) -> RewardFeeFraction {
+        self.reward_fee_fraction.clone()
+    }
+
+    /*******************/
+    /* Owner's methods */
+    /*******************/
+
+    /// Owner's method.
+    /// Updates current public key to the new given public key.
+    pub fn update_staking_key(&mut self, stake_public_key: Base58PublicKey) {
+        assert_eq!(
+            env::predecessor_account_id(),
+            self.owner_id,
+            "Can only be called by the owner"
+        );
+        self.ping();
+        self.stake_public_key = stake_public_key.into();
+        self.restake();
+    }
+
+    /// Owner's method.
+    /// Updates current reward fee fraction to the new given fraction.
+    pub fn update_reward_fee_fraction(&mut self, reward_fee_fraction: RewardFeeFraction) {
+        assert_eq!(
+            env::predecessor_account_id(),
+            self.owner_id,
+            "Can only be called by the owner"
+        );
+        reward_fee_fraction.assert_valid();
+
+        self.ping();
+        self.reward_fee_fraction = reward_fee_fraction;
+        self.maybe_restake();
+    }
+
+    /// Owner's method.
+    /// Vote on a given proposal with the given stake amount on a given voting contract account ID
+    /// on behalf of the pool.
+    /// NOTE: This method allows the owner to call `vote(proposal_id: U64, stake: U128)` on any
+    /// contract on behalf of this staking pool.
+    pub fn vote(
+        &mut self,
+        voting_account_id: AccountId,
+        proposal_id: ProposalId,
+        stake: U128,
```

Comment:
What is stake? How it's verified?

Reply:
Should be part of voting spec: https://github.com/nearprotocol/NEPs/pull/62